setup.py

```diff
@@ -300,7 +300,7 @@ def cmake(build_dir: str = "", install_prefix: str = "./nvfuser"):
 
     if not CMAKE_ONLY:
         # build binary
-        max_jobs = os.getenv("MAX_JOBS", str(multiprocessing.cpu_count()))
+        max_jobs = os.getenv("MAX_JOBS", min(str(multiprocessing.cpu_count()), "24"))
```

Comment:
Why?

Reply:
If some node is failing, should we just set `MAX_JOBS` for that node? 24 jobs is such a small number. On a 12 core CPU, there will be 24 threads. And we have pretty much nodes with > 12 cores.

Reply:
I think my suggestion for this is: for a machine that people use daily, people can just set `MAX_JOBS` manually in their `bashrc` and use whatever jobs they want so it does not really matter for this case. But when someone needs to go to a new environment (especially when it is inside a docker container), for example some shared node in company, usually in that environment, the `MAX_JOBS` will not be set. We should make the cap here larger so in our common machines, the compilation speed is maximum without manually setting `MAX_JOBS`. It is bad experience if everytime when you grab a node, you need to do a `lscpu` and check what is the cpu number and set `MAX_JOBS` manually.

Reply:
I don't mind merging this PR as is, but if this is the case, please do open a separate PR for more discussion. Also, should this be the following?
```suggestion
        max_jobs = os.getenv("MAX_JOBS", str(min(multiprocessing.cpu_count(), 24)))
```

Reply:
well, I guess we don't have to. it's killing our a100 node on nightly container.